adoc/chapters/programming_interface.adoc

```diff
@@ -14598,9 +14599,13 @@ This invocation function ignores any [code]#kernel_bundle# that was bound to
 this command group handler via [code]#handler::use_kernel_bundle()# and instead
 implicitly uses the kernel bundle that contains the [code]#kernelObject#.
 Throws an [code]#exception# with the [code]#errc::kernel_not_supported# error
-code if the [code]#kernelObject# is not compatible with either the device
-associated with the primary queue of the <<command-group>> or with the device
-associated with the secondary queue (if specified).
+code if the [code]#kernelObject# is not compatible with the device
+associated with the primary queue of the <<command-group>>.
+Throws an [code]#exception# with the [code]#errc::kernel_not_supported# error
+code if the <<command-group>> has a secondary queue associated that the
+implementation intends on using in the case that the <<command-group>> fails to
+enqueue on the primary queue, and the [code]#kernelObject# is not compatible
+with the device associated with the secondary queue of the <<command-group>>.
```

Comment:
For all these kernel invocation commands, we can shorten the description by using the new terminology:

> Throws an `exception` with the `errc::kernel_not_supported` error code if the \<\<command-group\>\> has a secondary queue, the implementation supports the secondary queue fallback, and the `kernelObject` is not compatible with the device associated with the secondary queue.